Client/U2FExtensions.swift

```diff
@@ -422,10 +423,17 @@ class U2FExtensions: NSObject {
         let authenticatorData = response.authData.base64EncodedString()
         let clientDataJSONString = clientDataJSON.base64EncodedString()
         let sig = response.signature.base64EncodedString()
+
+       let userHandle = response.user?.userId.base64EncodedString() ?? ""
+        
+        guard let requestId = response.credential?.credentialId.base64EncodedString() else {
+            sendFIDO2AuthenticationError(handle: handle)
+            return
+        }
         
         cleanupFIDO2Authentication(handle: handle)
         ensureMainThread {
-            self.tab?.webView?.evaluateJavaScript("navigator.credentials.postGet('\(handle)', \(true), '\(requestId)', '\(authenticatorData)', '\(clientDataJSONString)', '\(sig)', '')", completionHandler: { _, error in
+            self.tab?.webView?.evaluateJavaScript("navigator.credentials.postGet('\(handle)', \(true), '\(requestId)', '\(authenticatorData)', '\(clientDataJSONString)', '\(sig)', '\(userHandle)', '')", completionHandler: { _, error in
```

Comment:
are any of the parameters to `navigator.credentials.postGet` controllable by an external party (other than the user and the webpage that this JS is being evaluated on)? if so seems like there could be an XSS here (for instance set `clientDataJSONString` to `'); console.log(document.cookie);...` or something

Reply:
These values are not controlled by any external user. `postGet` creates an `AuthenticatorAssertionResponse` object where the `clientDataJSONString` is converted to an ArrayBuffer and returned to the caller, see here: https://github.com/brave/brave-ios/pull/1327/files#diff-dd925ee95e1f0ee3f47db69a4451d76aR91

Reply:
where are the inputs to the constructor passed in? https://github.com/brave/brave-ios/pull/1327/files#diff-dd925ee95e1f0ee3f47db69a4451d76aR89

Reply:
https://github.com/brave/brave-ios/pull/1327/files#diff-dd925ee95e1f0ee3f47db69a4451d76aR142

Reply:
cool, as long as you are confident this isn't an injection vector, lgtm